aptos-move/framework/aptos-framework/sources/transaction_validation.move

```diff
@@ -152,6 +173,22 @@ module aptos_framework::transaction_validation {
         prologue_common(sender, gas_payer, txn_sequence_number, txn_public_key, txn_gas_price, txn_max_gas_units, txn_expiration_time, chain_id)
     }
 
+    fun script_prologue_gas_deposit(
```

Comment:
hmm.. thought required deposit is always None for scripts?

Reply:
it's true in the randomness context. but a future feature may require it?

Reply:
this mechnism defends against bed users, not bad developers, and for a script the user can always remove your #[randomness] annotation before sending the txn. I feel before the semantics are clear we should not introduce the code -- can confuse dumb people like me who's reading the code.